### PR TITLE
chore(compass-aggregations): treat empty pipeline source as empty array COMPASS-6368

### PR DESCRIPTION
**Pull request by gribnoysup**
This removes the error and allows to run / export pipeline when input in the text editor is empty

### CODE REVIEW

File: packages/compass-aggregations/src/modules/pipeline-builder/pipeline-builder.ts [hunk 1]
```diff
@@ -17,28 +17,63 @@ export const DEFAULT_PIPELINE = `[]`;
...
   constructor(dataService: DataService, source = DEFAULT_PIPELINE) {
     this.previewManager = new PipelinePreviewManager(dataService);
-    this.source = source;
+    this.changeSource(source);
     this.sourceToStages();
   }
 
-  set source(source: string) {
-    this._source = source;
-    this.parseSourceToPipeline();
```
Comment (gribnoysup) [inline packages/compass-aggregations/src/modules/pipeline-builder/pipeline-builder.ts:37]: Drive-by: the logic was spread between `changeSource` method and setter, so sometimes we would set the value and then call the `validateSource` method that changeSource calls, I aligned everything to use `changeSource` because it's how we do this for stages and what is being used in the redux slice

**Review by mabaasit — APPROVED** [review]
